core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -116,11 +119,14 @@ public void initialize(String name, Map<String, String> properties) {
         fileIOImpl == null
             ? new HadoopFileIO(conf)
             : CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
+    this.encryptionManagerFactory = new DefaultEncryptionManagerFactory();
+    this.encryptionManagerFactory.initialize(properties);
```

Comment:
This pattern is typically used for dynamically loaded implementations. Since this is part of the catalog, I think it makes sense to just pass the properties to the encryption manager constructor.